### ISSUE TITLE
[BUG] Impossible to change timeframe window

### ISSUE THREAD

**Issue by lukou31** [henrywhitaker3/Speedtest-Tracker#680]
**Describe the bug**
No effect when changing "Show results for the last" to a different value from 7. Working fine before the upgrade.

**Context**
Version : 1.12.0
Synology - DSM7

**Comment by brommeltje** [comment]
Same issue on debian 10

**Comment by makitsune** [comment]
Same here, just running latest Docker image on x86_64

**Comment by jaxjexjox** [comment]
Ok so it's not just me!  Damnit.  I've logged a duplicate of this. Sorry.

**Comment by jaxjexjox** [comment]
Dupe here, sorry.
https://github.com/henrywhitaker3/Speedtest-Tracker/issues/682

**Comment by jaxjexjox** [comment]
Does anyone know how to pull an old version, this is occurring at the WORST possible time for me!  I need to provide data for a line fault to my ISP :(

I only see 4 versions here?! 
https://hub.docker.com/r/henrywhitaker3/speedtest-tracker/tags?page=1&ordering=-last_updated

Do I just pull dev?

**Comment by makitsune** [comment]
> Does anyone know how to pull an old version, this is occurring at the WORST possible time for me! I need to provide data for a line fault to my ISP :(
> 
> I only see 4 versions here?!
> https://hub.docker.com/r/henrywhitaker3/speedtest-tracker/tags?page=1&ordering=-last_updated
> 
> Do I just pull dev?

The dev doesn't upload each individual version on Docker hub for some reason... But I looked through my `docker images` and used the image id of the previously downloaded version

**Comment by jaxjexjox** [comment]
I have pulled the 4 month old 'dev' release and it has fixed nothing, the image appears to be identical somehow.
I (thought) I had a rough idea on how to work with docker, alas, perhaps this is not the case.

Anyone got any idea, how I can get back to the old edition, I need to convince my ISP I have a line fault and I'd really like to be able to produce some nice graphs.

**Comment by jaxjexjox** [comment]
> > Does anyone know how to pull an old version, this is occurring at the WORST possible time for me! I need to provide data for a line fault to my ISP :(
> > I only see 4 versions here?!
> > https://hub.docker.com/r/henrywhitaker3/speedtest-tracker/tags?page=1&ordering=-last_updated
> > Do I just pull dev?
> 
> The dev doesn't upload each individual version on Docker hub for some reason... But I looked through my `docker images` and used the image id of the previously downloaded version

I'm an idiot who cleans up disk space by purging and pruning old stuff! :(

Can they sill be pulled?  Please let me know the tag so i can spin up a working one of these, to show first level internet support (!! :( :( .....!!!!)  a nice simple graphic, clearly showing my new line speed fault.

**Comment by makitsune** [comment]
I've uploaded a tar.gz of the image on my server here: https://files.cutelab.space/f/ebb317bf98074fe78340/
Once loaded, you can use it with image id: 25b86b3d603a

I don't know how to prove this is the official release... I just want to help out. The owner of this project should really push the version tag and then latest tag to Docker hub.

```bash
wget -O speedtest-tracker-1.10.4.tar.gz "https://files.cutelab.space/f/ebb317bf98074fe78340/?dl=1"
docker load -i speedtest-tracker-1.10.4.tar.gz
```

**Comment by makitsune** [comment]
You might have to backup your data in the settings first, then remove your data volume, start the container fresh and import your data back in. It will break if you try to downgrade.

**Comment by jaxjexjox** [comment]
I <3 linux, open source stuff, docker and you!

Magic, just wonderful.
Thanks for the docker load command, I sat here in my windows mindset wondering how to extract the tar, copy the files into a folder full of images to work with.  You probably saved me hours.

Did indeed have to backup data, been a bit of a mess but I have graphs now to show my ISP my line is stuffed.
Thanks again.
MUST stop being in the habit of purging and pruning behind me so regularly.

**Comment by makitsune** [comment]
I'm really glad I could help! I prune things pretty often too and I'm not afraid of going cutting edge. But this project definitely needs a Docker overhaul. It's not done correctly. Not only do versions not get released, but the owner copies the root of the main branch into a separate branch in a weird way, compile on their machine and then build the image.

I might send in a pr so you can `docker build` and it will compile and put everything together like it should. Then you can make fresh builds of any version without understanding php or npm.

**Comment by jaxjexjox** [comment]
They probably don't know any better (I know I don't) so any help for the original developer is probably quite helpful.

The only reason I'm using this, is because this guy broke this one.
https://github.com/roest01/docker-speedtest-analyser/issues/75

It need a fork or some such which updates a file.
https://github.com/roest01/docker-speedtest-analyser/pull/82

**Comment by Sn00zEZA** [comment]
Also having this issue

**Comment by henrywhitaker3** [comment]
Building a fix for this now lads. Was an issue with the refresh happening before the state was change, if u update the number of days again it has the right value the second time

**Comment by henrywhitaker3** [comment]
Right turns out there was another issue, but have pushed a new `dev` and `dev-arm` image - see if it sorts it for you and will merge to master

**Comment by db77x** [comment]
dev build 12.1 working now for down/upload and ping. But not for successful/faild graph. 
Testest with various days 30, 180, 360 etc. 
Maybe you can fix it also @henrywhitaker3 ?

**Comment by burtaverde** [comment]
can confirm, same results as db77x. Also i might be wrong, but if i remember correctly, it used to show the timeline on all graphs and currently they're empty on the bottom

**Comment by masterwishx** [comment]
confirm (dev) working Download,Upload,Ping Graphs on Unraid (linux/amd64) but not the Failure graph

**Comment by masterwishx** [comment]
v 1.12.2  dev all graphs is working now in Unraid (linux/amd64)

**Comment by jaxjexjox** [comment]
I take it this one isn't published as official yet, right?  (where can I check? Sorry for hassling!)

**Comment by jaxjexjox** [comment]
Can I just confirm, this is still broken, in both latest and dev docker builds, right? (It's still dead for me)

**Comment by tuckshoprn** [comment]
It is working for me on the dev branch. Running in docker on Linux server.

**Comment by jaxjexjox** [comment]
Ok that's a bit odd,  perhaps I need to nuke my data directory.    I also notice this one sure chews some disk space.

**Comment by jaxjexjox** [comment]
Yeah looks like I needed a fresh data directory, no biggie, I could do with a fresh folder anyhow.  Thanks for hard work henry.

**Comment by tuckshoprn** [comment]
Mine is around 337MB which includes 2.7MB .db file.

**Comment by jaxjexjox** [comment]
I mean look, disk space is cheap, I'll get over it, but I'll say the CSV file from other apps is in the kilobytes, I'm not sure why so much is on the disk but no big deal, at least this works!  Unlike roests one.

**Comment by Sn00zEZA** [comment]
When will this bug fix merge with latest for docker?
